kobo/django/upload/models.py

```diff
@@ -54,6 +54,9 @@ def get_full_path(self):
     def __unicode__(self):
         return six.text_type(os.path.join(self.target_dir, self.name))
 
+    def __str__(self):
+        return six.text_type(os.path.join(self.target_dir, self.name))
```

Comment:
Same here about str vs unicode.  On Python2, this is supposed to return a `str` and not a `unicode`.  On Python3 it's also supposed to return a `str`, so there's no need for six at all here:

```suggestion
        return str(os.path.join(self.target_dir, self.name))
```